### PR TITLE
ROC-6926: Add mediation opt in app insights

### PR DESCRIPTION
**Pull request by dharmendrak**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/ROC-6926

### Change description ###

ROC-6926: Add mediation opt in app insights

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationSuccessCallbackHandler.java  1
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationSuccessCallbackHandlerTest.java  1
- domain-model/src/main/java/uk/gov/hmcts/cmc/domain/utils/FeaturesUtils.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/cmc-claim-store/cmc-claim-store/pullRequest?prid=4516348&bid=15215796)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationSuccessCallbackHandlerTest.java  1
- domain-model/src/main/java/uk/gov/hmcts/cmc/domain/utils/FeaturesUtils.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationSuccessCallbackHandler.java  1
         

Complexity decreasing per file
==============================
+ src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimantResponseService.java  -3
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/cmc-claim-store/cmc-claim-store/pullRequest?prid=4516348&bid=15215796)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- domain-model/src/main/java/uk/gov/hmcts/cmc/domain/utils/FeaturesUtils.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationSuccessCallbackHandler.java  1
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationSuccessCallbackHandlerTest.java  1
         

Complexity decreasing per file
==============================
+ src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimantResponseService.java  -3
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/cmc-claim-store/cmc-claim-store/pullRequest?prid=4516348&bid=15215796)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationSuccessCallbackHandler.java  2
- domain-model/src/main/java/uk/gov/hmcts/cmc/domain/utils/FeaturesUtils.java  1
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationSuccessCallbackHandlerTest.java  1
         

Complexity decreasing per file
==============================
+ src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimantResponseService.java  -3
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/cmc-claim-store/cmc-claim-store/pullRequest?prid=4516348&bid=15215796)

File: domain-model/src/main/java/uk/gov/hmcts/cmc/domain/utils/ResponseUtils.java [hunk 1]
```diff
@@ -84,8 +86,12 @@ public static boolean isFullDefence(Response response) {
...
     }
 
-    public static boolean isNoMediation(Response response) {
-        return response.getFreeMediation().filter(Predicate.isEqual(YesNoOption.NO)).isPresent();
+    public static boolean defendantNotOptedForMediation(Response response) {
+        return response.getFreeMediation().filter(Predicate.isEqual(NO)).isPresent();
+    }
+
+    public static boolean defendantOptedForMediation(Response response) {
+        return response.getFreeMediation().filter(Predicate.isEqual(YES)).isPresent();
```
Comment (KiranVarma02) [inline domain-model/src/main/java/uk/gov/hmcts/cmc/domain/utils/ResponseUtils.java]: can it be 1 method and called something like hasDefendantOptedForMediation?

Comment (dharmendrak) [reply domain-model/src/main/java/uk/gov/hmcts/cmc/domain/utils/ResponseUtils.java]: actioned

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationSuccessCallbackHandlerTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationSuccessCallbackHandler.java  2
- domain-model/src/main/java/uk/gov/hmcts/cmc/domain/utils/FeaturesUtils.java  1
         

Complexity decreasing per file
==============================
+ src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimantResponseService.java  -3
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/cmc-claim-store/cmc-claim-store/pullRequest?prid=4516348&bid=15215796)

**Pull request merged**